src/backend/renderer/multigpu/mod.rs

```diff
@@ -0,0 +1,1755 @@
+    <T::Device as ApiDevice>::Renderer: ImportDma + ImportMem,
+    <<R::Device as ApiDevice>::Renderer as Renderer>::Error: 'static,
+    <<T::Device as ApiDevice>::Renderer as Renderer>::Error: 'static,
+{
+    type Error = Error<R, T>;
+    type TextureId = MultiTexture;
+    type Frame = MultiFrame<R, T>;
+
+    fn id(&self) -> usize {
+        self.render.renderer().id()
+    }
+
+    fn downscale_filter(&mut self, filter: TextureFilter) -> Result<(), Self::Error> {
+        self.render
+            .renderer_mut()
+            .downscale_filter(filter)
+            .map_err(Error::Render)
+    }
+    fn upscale_filter(&mut self, filter: TextureFilter) -> Result<(), Self::Error> {
+        self.render
+            .renderer_mut()
+            .upscale_filter(filter)
+            .map_err(Error::Render)
+    }
+
+    fn render<F, Res>(
+        &mut self,
+        size: Size<i32, Physical>,
+        dst_transform: Transform,
+        rendering: F,
+    ) -> Result<Res, Self::Error>
+    where
+        F: FnOnce(&mut Self, &mut Self::Frame) -> Res,
+    {
+        let buffer_size = size.to_logical(1).to_buffer(1, dst_transform);
+        if self.target.is_some() {
+            let render_buffer = Offscreen::<Target>::create_buffer(self.render.renderer_mut(), buffer_size)
+                .map_err(Error::Render)?;
+            self.render
+                .renderer_mut()
+                .bind(render_buffer)
+                .map_err(Error::Render)?;
+        }
+
+        let node = *self.render.node();
+
+        // we need to move some stuff into the closure temporarily
+        let mut dma_source = self.dma_source.take();
+        let mut target = self.target.take();
+        let mut other_renderers = self.other_renderers.drain(..).collect::<Vec<_>>();
+        let dma_source_ref = &mut dma_source;
+        let target_ref = &mut target;
+        let other_renderers_ref = &mut other_renderers;
+
+        let log = self.log.clone();
+        let res = self
+            .render
+            .renderer_mut()
+            .render(size, dst_transform, move |render, frame| {
+                let mut new_renderer = MultiRenderer {
+                    dma_source: dma_source_ref.take(),
+                    render: RenderDevice::Renderer(render, node),
+                    target: target_ref.take(),
+                    other_renderers: other_renderers_ref.drain(..).collect(),
+                    proxy_framebuffer: std::marker::PhantomData,
+                    log: log.clone(),
+                };
+                let mut frame = MultiFrame {
+                    node,
+                    frame, // we cheat here and use a raw-ptr, because otherwise your associated type would gain an uncostraint lifetime parameter
+                    damage: Vec::new(),
+                    _target: std::marker::PhantomData::<T>,
+                    log,
+                };
```

Comment:
This should receive a `FIXME` for when GATs are released if this is the only real solution for now.

Reply:
Added FIXME to MultiFrame struct